test/protobuf/text_test.exs

```diff
@@ -0,0 +1,150 @@
+defmodule Protobuf.TextTest do
+  use ExUnit.Case, async: true
+
+  alias Protobuf.Text
+
+  test "default fields aren't encoded" do
+    # proto3
+    assert "" == Text.encode(%TestMsg.Foo{})
+
+    # proto2
+    assert "a: 1" == Text.encode(%TestMsg.Foo2{a: 1})
+  end
+
+  test "encoding basic types" do
+    assert ~S(a: 1, b: "foo") == Text.encode(%TestMsg.Foo.Bar{a: 1, b: "foo"})
+  end
+
+  test "encoding enums" do
+    assert ~S(j: D) == Text.encode(%TestMsg.Foo{j: :D})
+  end
+
+  test "encoding repeated" do
+    assert ~S(g: [1, 2, 3, 4, 5, 6]) == Text.encode(%TestMsg.Foo{g: [1, 2, 3, 4, 5, 6]})
+  end
+
+  test "encoding large repeated breaks lines" do
+    result = Text.encode(%TestMsg.Foo{a: 1, g: List.duplicate(1_111_111_111, 7), j: :D})
+
+    assert result == """
+           a: 1
+           g: [
+             1111111111,
+             1111111111,
+             1111111111,
+             1111111111,
+             1111111111,
+             1111111111,
+             1111111111
+           ]
+           j: D\
+           """
+  end
+
+  test "encoding nested structs" do
+    result = Text.encode(%TestMsg.Foo{e: nil})
+    assert result == ~S()
+
+    result = Text.encode(%TestMsg.Foo{e: %TestMsg.Foo.Bar{}})
+    assert result == ~S(e: {})
+
+    result = Text.encode(%TestMsg.Foo{e: %TestMsg.Foo.Bar{a: 1, b: "Hello"}})
+
+    assert result == ~S(e: {a: 1, b: "Hello"})
+  end
+
+  test "encoding large nested structs" do
+    result =
+      Text.encode(%TestMsg.Foo{
+        a: 1,
+        e: %TestMsg.Foo.Bar{a: 1, b: String.duplicate("Hello", 15)},
+        h: [
+          %TestMsg.Foo.Bar{a: 5},
+          %TestMsg.Foo.Bar{a: 1, b: String.duplicate("Hello", 15)},
+          %TestMsg.Foo.Bar{a: 7}
+        ]
+      })
+
+    assert result == """
+           a: 1
+           e: {
+             a: 1,
```

Comment:
I'm not very happy with this inconsistency: notice that the "outside" entries don't have `,`, but the nested ones do. Tried to make it match but failed. 

This is fine per the spec.